Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-ExecutionTimeAnomaly.yaml

```diff
@@ -0,0 +1,75 @@
+id: af55d5b0-6b4a-4874-8299-9d845bf7c1fd
+name: Anomalous Query Execution Time
+description: |
+  'This hunting query will detect SQL queries that took an unsually long period of time to execute based on a calculated average execution time. The query groups based on the application and the username, making this query sutable for detecting exploitation of web applications, or other SQL backed applications with predictable behaviour.'
+requiredDataConnectors: []
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1190
+tags:
+  - SQL
+query: |
+
+  let timeRange = 14d;
+  //How frequently the query averages data for an average execution time
+  let timeSliceSize = 1h;
+  //Anomaly decompose threshold, 2 by default
+  let scoreThreshold = 2;
+  let processedData = materialize (
+      AzureDiagnostics
+      | where TimeGenerated > ago(timeRange)
+      | where Category == 'SQLSecurityAuditEvents' and action_id_s has_any ("RCM", "BCM") // Keep only SQL affected rows
+      | project TimeGenerated, PrincipalName = server_principal_name_s, ClientIp = client_ip_s, HostName = host_name_s, ResourceId,
+      ApplicationName = application_name_s, ActionName = action_name_s, Database = strcat(LogicalServerName_s, '/', database_name_s),
+      IsSuccess = succeeded_s, DurationMs = duration_milliseconds_d, AffectedRows = affected_rows_d,
+      ResponseRows = response_rows_d, Statement = statement_s,
+      Error = case( additional_information_s has 'error_code', toint(extract("<error_code>([0-9.]+)", 1, additional_information_s))
+                        , additional_information_s has 'failure_reason', toint(extract("<failure_reason>Err ([0-9.]+)", 1, additional_information_s))
+                        , 0),
+      State = case( additional_information_s has 'error_state', toint(extract("<error_state>([0-9.]+)", 1, additional_information_s))
+                        , additional_information_s has 'failure_reason', toint(extract("<failure_reason>Err ([0-9.]+), Level ([0-9.]+)", 2, additional_information_s))
+                        , 0),
+      AdditionalInfo = additional_information_s, timeSlice = floor(TimeGenerated, timeSliceSize));
+  processedData
+  //Bin the data into 1h windows, taking the average of exeuction time
+  | summarize round(avg(DurationMs), 2), min(TimeGenerated), max(TimeGenerated) by PrincipalName, ApplicationName, bin(TimeGenerated, timeSliceSize)
+  //Summarise by user and application and create list ready for anomaly detection
+  | summarize make_list(avg_DurationMs), make_list(min_TimeGenerated), make_list(max_TimeGenerated) by PrincipalName, ApplicationName
+  | extend series_decompose_anomalies(list_avg_DurationMs, scoreThreshold, -1, 'linefit')
+  | mv-expand TimeAnomaly=series_decompose_anomalies_list_avg_DurationMs_ad_flag, WindowStart=list_min_TimeGenerated, WindowEnd=list_max_TimeGenerated
+  | project WindowStart, WindowEnd, PrincipalName, ApplicationName, TimeAnomaly
```

Comment:
ResoureId is dropped in this line and the query will fail later on the last project statement as ResourceId is not brought through the left side of the join.

Reply:
This is not resolved, The project statement on line 11 doesn't have ResourceId and so it fails later.